### ISSUE TITLE
Leverage or consider using LinePlot from Bamboo

### ISSUE THREAD

**Issue by veillette** [phetsims/calculus-grapher#54]
CurveNode from CG create custom plots. We could consider using LinePlot from Bamboo. 

Since LinePlot uses ChartTransform as an argument, it would make it very easy to scale our graph.
On the other hand, CG is using its own custom data field whereas LinePlot expects an array of Vector2 points.

**Comment by veillette** [comment]
CurveNode creates a Path of CurvePoints and well as an array of Circles  representing Cusps. I suggest to replace the Path by LinePlot and The array of circle by a ScatterPlot.

We will need to process the Curve to generate  a data set that is appropriate to the LinePlot ad ScatterPlot. 

There will be no need for the ModelViewTransform and its property once we make the conversion since LinePlot and ScatterPlot uses a ChartTransform instead. 

It is not clear what the impact will be on performance.

**Comment by veillette** [comment]
I committed some preliminary work. Using Bamboo simplify our approach, but there seems to be some performance tradeoff.

Some of the performance tradeoff may be related to the way that the various DataSet are updated where new Vector allocations are created for each update.

In addition, and most importantly, I wasn't able to attach a listener directly to LinePath. Instead, as a placeholder I attached the listener onto a scatterPlot  that mimics the linePlot instead. I dont understand why my initial approach did not work.
